### ISSUE TITLE
N805: False positive with @abstractclassmethod

### ISSUE THREAD

**Issue by mflova** [astral-sh/ruff#5307]
Using the latest release.

Minimal example:

```python
from abc import ABC, abstractclassmethod

class Rule(ABC):
    @abstractclassmethod
    def check(cls) -> None:
        ...
```

It forces me to rename it to `self` because it is not able to detect the pressence of `abstractclassmethod`

This one seems to be deprecated since Python 3.3 so I am not sure if it makes sense to fix it

**Comment by mflova** [comment]
My bad. I missinterpret the decorator

**Comment by charliermarsh** [comment]
I think it's reasonable to fix this, even if deprecated.

**Comment by lightswitch05** [comment]
It seems this fix only works if using `@abc.abstractclassmethod` instead of `@abstractclassmethod`?
